### PR TITLE
Adding more clear dependencies list, target audience, learning objectives and statement of need

### LINKED ISSUE #19: [JOSE review]: Statement of need
The section in the paper titled "statement of need" doesn't quite state the need for these materials. Maybe explain what kinds of gaps the course from which these materials were derived filled in the curriculum that this was part of?

### LINKED ISSUE #22: [JOSE review]: List of software dependencies
Though there are local references to dependencies in various parts of the materials (including in slides and in videos) It would be good to have a list of all the software dependencies in one place (e.g., in the README file).

### PR DESCRIPTION
**Pull request by kmader**
Adding a complete list of software dependencies for reproducibility.
- closes #22 #21 #20 
- might close #19

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,10 +5,28 @@
...
+
+# Overview
+
+The lecture focuses on the challenging task of extracting robust, quantitative metrics from imaging data and is intended to bridge the gap between pure signal processing and the experimental science of imaging. The course will focus on techniques, scalability, and science-driven analysis.
+
+## Learning Objectives
+
+1. Introduction of applied image processing for research science covering basic image processing, quantitative methods, and statistics.
+2. Understanding of imaging as a means to accomplish a scientific goal. 
+3. Ability to apply quantitative methods to complex 3D data to determine the validity of a hypothesis
```
Comment (arokem) [inline README.md]: These are too general to be learning objectives. Take a look at the explanation [here](http://swcarpentry.github.io/swc-releases/2016.06/instructor-training/20-lessons/). Objectives should be "specific, verifiable descriptions of what learners can do to demonstrate their learning".

File: README.md [hunk 2]
```diff
@@ -5,10 +5,28 @@
...
+
+## Target Audience
+
+Ideally students will have some familiarity with basic manipulation and programming in languages like Python (Matlab or R are also reasonable starting points). Much of the material is available as visual workflows in a tool called KNIME, although these are less up to date as the Python material. Interested students who are worried about their skill level in this regard are encouraged to contact Kevin Mader directly (mader@biomed.ee.ethz.ch).
```
Comment (arokem) [inline README.md]: Undergraduate? Graduate? In what fields?

**Comment by kmader** [comment]
I have the statement of need in the [paper.md](https://github.com/kmader/Quantitative-Big-Imaging-2018/blob/software-deps/paper.md#statement-of-need), should I also have it in the readme?

File: README.md (hunk 1 above)
Comment (kmader) [reply README.md]: thanks as I try to write better and clearer objectives I begin to appreciate that having these clearly stated would probably have helped students in preparing for exams

**Pull request merged**